### PR TITLE
[DOM-120] Add GET device token information contract

### PR DESCRIPTION
**Pull request by ms-darianbenito**
Related to [DOM-120](https://makingsense.atlassian.net/browse/DOM-120)

This PR is to create a GET endpoint to obtain device token information. This PR has no endpoint implementation, just contract definition.

### CODE REVIEW

File: Doppler.Push.Api/Controllers/DeviceTokenController.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+namespace Doppler.Push.Api.Controllers
+{
+    [Authorize(Policies.ONLY_SUPERUSER)]
+    [ApiController]
+    public class DeviceTokenController : ControllerBase
```
Comment (andresmoschini) [inline Doppler.Push.Api/Controllers/DeviceTokenController.cs:12]: Why do you use the `ControllerBase`? What do you think about simply using POCO controllers?

File: Doppler.Push.Api/Controllers/DeviceTokenController.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    {
+        [HttpGet]
+        [Route("device-tokens/{deviceToken}")]
+        public async Task<ActionResult<DeviceTokenInfo>> GetInfo([FromRoute] string deviceToken)
```
Comment (andresmoschini) [inline Doppler.Push.Api/Controllers/DeviceTokenController.cs:16]: Why do you have a resource to represent the _token of a device_, in place of having a resource to represent the _device_ himself?

```suggestion
        [HttpGet]
        [Route("devices/{deviceToken}")]
        public async Task<ActionResult<Device>> Get([FromRoute] string deviceToken)
```

Comment (ms-darianbenito) [reply Doppler.Push.Api/Controllers/DeviceTokenController.cs:16]: Great idea! Thanks

File: Doppler.Push.Api/Controllers/DeviceTokenController.cs (hunk 1 above)
Comment (ms-darianbenito) [reply Doppler.Push.Api/Controllers/DeviceTokenController.cs:12]: I extend `ControllerBase` to use methods like [`BadRequest()`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.controllerbase.badrequest?view=aspnetcore-5.0). But, maybe I can create POCO controllers and use, for example, `BadRequestResult()` instead.

With `ControllerBase`:
```
{
    return BadRequest();
}
```

POCO controller:
```
{
    return new BadRequestResult();
}

```

Without  `ControllerBase` I just use what I need. Thanks

File: Doppler.Push.Api/Controllers/DeviceTokenController.cs (hunk 2 above)
Comment (ms-darianbenito) [reply Doppler.Push.Api/Controllers/DeviceTokenController.cs:16]: @andresmoschini I am thinking on change `deviceToken` route param to `token`:
```suggestion
        [HttpGet]
        [Route("devices/{token}")]
        public async Task<ActionResult<Device>> Get([FromRoute] string token)
```
`Device` model will be:
```
public class Device
    {
        public string Token { get; set; }

        public bool IsValid { get; set; }
    }
```

Can you give me your opinion?

File: Doppler.Push.Api/Controllers/DeviceTokenController.cs (hunk 1 above)
Comment (andresmoschini) [reply Doppler.Push.Api/Controllers/DeviceTokenController.cs:12]: I like POCO, but it is opinable.

File: Doppler.Push.Api/Controllers/DeviceTokenController.cs (hunk 2 above)
Comment (andresmoschini) [reply Doppler.Push.Api/Controllers/DeviceTokenController.cs:16]: Looks nice!

File: Doppler.Push.Api/Controllers/DeviceTokenController.cs (hunk 1 above)
Comment (ms-darianbenito) [reply Doppler.Push.Api/Controllers/DeviceTokenController.cs:12]: I will use POCO in this controller. I am going to see if time makes me miss `ControllerBase` :P

**Pull request merged**